homeassistant/components/alert/__init__.py

```diff
@@ -307,7 +303,7 @@ async def _send_notification_message(self, message):
 
         for target in self._notifiers:
             await self.hass.services.async_call(
-                DOMAIN_NOTIFY, target, msg_payload, context=self._context
+                Platform.NOTIFY, target, msg_payload, context=self._context
```

Comment:
I don't agree with this. A service call is domain-based, not platform-based.

Multiple occurrences in this PR.

Reply:
Hmm 🤔
I, maybe incorrectly (?), assumed that we wanted to replace all `DOMAIN` constants from the default platforms with `Platform.xxx`. Your concerns are valid though, the question is what to do now.

Personally, I would like to remove the `DOMAIN` constants, at least the "default" once as they can quite easily create a confusion with the integration domain constant. Should we consider creating another StrEnum for Domains which are commonly used in service calls? That feels a bit redundant tbh.

What is your opinion on it?

Reply:
I follow up on a comment, I believe, @MartinHjelmare placed about this in other PRs.

Originally the `Platform` enum was created for a specific case (platforms list in integrations), however, it is already beginning to have it own life a bit.

I kinda agree with you, as in, use `Platform` for all platform integration domains everywhere a platform domain is used, is simple and consistent: In both review and the use as a developer.

Completely clean: 🤷 Not really I guess...

@MartinHjelmare ?

Reply:
I pushed for the Platform enum... so I guess I'm at least partly to blame for the confusion.
I think some arguments in some methods make it double hard to know if platform != domain != component. I'll find some sample methods tomorrow to show that.
Also it might be useful to refer to the arch discussion.

Reply:
> I kinda agree with you, as in, use `Platform` for all platform integration domains everywhere a platform domain is used, is simple and consistent: In both review and the use as a developer.
>
> Completely clean: 🤷 Not really I guess...

If the use case itself is ok and only the name `Platform` is mismatch, should we maybe try to come up with a better one? Just to suggest one, what about `PlatformDomain`? It's just a bit long tbh.

There are a few other domains which could make sense to add there if it's more general. I noticed these while working on the current changes, especially for the Google and Alexa components. Basically those domains that do have at least some use outside of their integration.

```py
group
input_boolean
input_button
input_select
script
select
zone
```

If a new name and / or additional domain values are an idea, I could open a draft PR where we could continue the discussion and maybe even find a better name. In any case, we would need a deprecation period for the "old" `Platform`.

Reply:
No, I think you misunderstand me. I think it's ok to keep `Platform` enum as it is.
There are a few places where the use of Platform enum could be recommended (eg. `config_entries.async_setup_platform` could have `Platform | str`)

I think we should add a distinct enum for when a domain is required (eg. `CommonDomain`)

My main issue is with some of the argument names.
Take `discovery.load_platform`
https://github.com/home-assistant/core/blob/395093351428d349246e4c1de526b159a167f382/homeassistant/helpers/discovery.py#L127-L133

In this context, if my understanding is right, `component` argument could be `Platform | str`, whereas `platform` argument would be the DOMAIN of the integration being loaded. And I think we need to sort out this type of confusion before we add a new enum.

Reply:
Maybe that would make more sense as:
```python
def load_platform( 
     hass: core.HomeAssistant, 
     platform: Platform | str, 
     integration_domain: str, 
     discovered: DiscoveryInfoType, 
     hass_config: ConfigType, 
 ) -> None: 
```

Reply:
The enum has a meaning and is connected with certain APIs. We can't use the enum just because an enum string matches another string. If we don't follow this, the point of the enum is lost.

The existing Platform enum should only be used for platform APIs where an entity platform is required, in my opinion. We should probably list these APIs to make that clear.

I earlier thought that some entity APIs like entity registry would also be ok for using the Platform enum, but I don't think so anymore since an entity can be registered without an entity platform, eg the person entity.

Reply:
I have created PR #63571 for `load_platform`/`async_load_platform`

Reply:
> The enum has a meaning and is connected with certain APIs. We can't use the enum just because an enum string matches another string. If we don't follow this, the point of the enum is lost.
>
> The existing Platform enum should only be used for platform APIs where an entity platform is required, in my opinion. We should probably list these APIs to make that clear.

Ok, makes sense. Let's continue in the direction @epenet proposed then with #63571. Going to close this and the other PRs like it.